README.md

```diff
@@ -1,2 +1,76 @@
 # comp110-worksheet-5
 Base repository for COMP110 worksheet 5
+
+
+### a.  
+This algorithm uses for loops to go through a list and check for duplicates.   
+n is the length of the list, i and j equal each variable in the list and it uses  
+an if statement to check if the variable in i and j are not equal to eachother.  
+This means if it goes through the list and i and j are not equal to eachother it  
+returns true, whereas if they it returns false.  
+  
+### b.  
+The worst case running time of this algorithm is O(n2) is because it involves  
+two for loops where one is nested in the other. This means the time complexity is times itself leading  
+to the n sqaured. Due to this with two for loops the slowest case of running time will be quadratic   
+represented as O(n2) in big O notation.  
+   
+To double check my awnser I used this archived paper from western science.  
+http://www.csd.uwo.ca/courses/CS1027b/notes/AnalysisIntro.pdf  
+  
+  
+### c.  
+The algorithm is still correct as it still uses a for loop to check through each list   
+and will still return true if each list doesnt have duplicates and still return false if  
+they do.    
+  
+### d.   
+The algorithm will run much faster, as when an algorithm is quadratic the time taken to   
+run increases rapidly for how much data it has to process. This is because the time complexity  
+grows quadratiaclly as n increases. When the change is made to  
+line 4 it has to process less data in the loops leading to a much faster runtime.  
```

Comment:
The key thing to notice is that the original algorithm checks each pair of elements twice (e.g. checks both `i,j` and `j,i`), whereas the modified algorithm checks each pair only once.